.travis.yml

```diff
@@ -54,9 +63,34 @@ script:
   - if [ "$DOCKER" == true ]; then bash ./scripts/travis/build-docker-images.sh ; else echo 'skipping docker images'; fi
   - if [ "$ES_INTEGRATION_TEST" == true ]; then bash ./scripts/travis/es-integration-test.sh ; else echo 'skipping elastic search integration test'; fi
   - if [ "$HOTROD" == true ]; then bash ./scripts/travis/hotrod-integration-test.sh ; else echo 'skipping hotrod example'; fi
+  - if [ "$LINUX" == true ]; then make build-linux ; else echo 'skipping linux'; fi
+  - if [ "$WINDOWS" == true ]; then make build-windows ; else echo 'skipping windows'; fi
+  - if [ "$DARWIN" == true ]; then make build-darwin ; else echo 'skipping darwin'; fi
```

Comment:
overall this looks good, but this 3 additions concern me as they are going to significantly increase Travis build times. I'd rather have a single DEPLOY=true step that would build files for all targets, which would be a lot faster since we don't have to wait for Travis VM/container to initialize.

Also, there's now a huge amount of duplication in the make file, I think it could be simplified a lot through variables, similar to the deploy.sh file, e.g.
```
linux-binaries: 
    make build-agent build-collector ... GOOS=darwin

build-agent:
    CGO_ENABLED=0 GOOS=$(GOOS) ...
```

Reply:
in fact, DEPLOY=true doesn't even need to be a new matrix entry, we can co-locate it with the DOCKER=true entry (since it requires linux binaries anyway).

Reply:
Thanks for the input @yurishkuro! I'll have time closer to the middle of this week to make these changes.